File: Jump-Game-II/Jump-Game-II.py


class Solution:
    # @param A, a list of integers
    # @return an integer
    def jump(self, A):
        n=len(A)
        if n==1:
            return 0
        maxdis=0
        ans=0
        while True:
            ans+=1
            for i in range(maxdis+1):
                maxdis=max(maxdis,A[i]+i)
                if maxdis>=n-1:
                    return ans
 

File: Jump-Game-II/Jump-Game-II.cpp

class Solution {
public:
    int jump(int A[], int n) {
        int ans=0;
        int last=0;
        int right=0;
        for(int i=0;i<n;i++)
        {
            if(i>last)
            {
                ans++;
                last=right;
            }
            right=max(right,i+A[i]);
        }
        return ans;
    }
};


 